python/plugins/processing/algs/qgis/EquidistanceBuffer.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+
+"""
+***************************************************************************
+    EquidistanceBuffer.py
+    --------------------
+    Date                 : April 2019
+    Copyright            : (C) 2019 by Ujaval Gandhi
+    Email                : ujaval at spatialthoughts dot com
+***************************************************************************
+*                                                                         *
+*   This program is free software; you can redistribute it and/or modify  *
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************
+"""
+
+__author__ = 'Ujaval Gandhi'
+__date__ = 'April 2019'
+__copyright__ = '(C) 2019, Ujaval Gandhi'
+
+# This will get replaced with a git SHA1 when you do a git archive323
+
+__revision__ = '$Format:%H$'
+
+from qgis.core import (QgsGeometry,
+                       QgsPoint,
+                       QgsFeature,
+                       QgsWkbTypes,
+                       QgsProcessing,
+                       QgsProcessingParameterDistance,
+                       QgsProcessingParameterNumber,
+                       QgsProcessingParameterEnum,
+                       QgsProcessingException,
+                       QgsCoordinateReferenceSystem,
+                       QgsCoordinateTransform,
+                       QgsProject
+                       )
+    
+from processing.algs.qgis.QgisAlgorithm import QgisFeatureBasedAlgorithm
+
+
+class EquidistanceBuffer(QgisFeatureBasedAlgorithm):
+    """
+    This algorithm creates equidistance buffers for vector layers.
+    
+    Each geometry is transformed to a Azimuthal Equidistant projection centered at that geometry,
+    buffered and transformed back to the original projection.
+    """
+    DISTANCE = 'DISTANCE'
+    SEGMENTS = 'SEGMENTS'
+    END_CAP_STYLE = 'END_CAP_STYLE' 
+    JOIN_STYLE = 'JOIN_STYLE'
+    MITER_LIMIT = 'MITER_LIMIT'
+
+    def group(self):
+        return self.tr('Vector geometry')
+
+    def groupId(self):
+        return 'vectorgeometry'
+
+    def __init__(self):
+        super().__init__()
+        self.distance = None
+        self.segments = None
+        self.join_style = None
+        self.side = None
+        self.miter_limit = None
+        self.join_styles = [self.tr('Round'),
+                            self.tr('Miter'),
+                            self.tr('Bevel')]
+        self.end_cap_styles = [self.tr('Round'),
+                       self.tr('Flat'),
+                       self.tr('Square')]
+
+    def initParameters(self, config=None):
+        self.addParameter(QgsProcessingParameterDistance(self.DISTANCE,
```

Comment:
You can call setDefaultUnit on this parameter with a unit type of meters to improve the ux

Reply:
Nice! I was annoyed by the <unknown> label. This is much better. I have added it.